src/Plugins/ComplexCore/test/RawBinaryReaderTest.cpp

```diff
@@ -0,0 +1,148 @@
+
+  // Instantiate the filter, a DataStructure object and an Arguments Object
+  RawBinaryReaderFilter filter;
+  DataStructure ds;
+  Arguments args;
+
+  // Test empty input file
+  std::string inputFile;
+
+  args.insertOrAssign(RawBinaryReaderFilter::k_InputFile_Key, std::make_any<FileSystemPathParameter::ValueType>(fs::path(inputFile)));
+  args.insertOrAssign(RawBinaryReaderFilter::k_ScalarType_Key, std::make_any<NumericType>(NumericType::uint8));
+  args.insertOrAssign(RawBinaryReaderFilter::k_NumberOfComponents_Key, std::make_any<int32>(1));
+  args.insertOrAssign(RawBinaryReaderFilter::k_Endian_Key, std::make_any<ChoicesParameter::ValueType>(0));
+  args.insertOrAssign(RawBinaryReaderFilter::k_SkipHeaderBytes_Key, std::make_any<uint64>(0));
+  args.insertOrAssign(RawBinaryReaderFilter::k_CreatedAttributeArrayPath_Key, std::make_any<DataPath>(createdAttributeArrayPathValue));
+
+  auto preflightResult = filter.preflight(ds, args);
+  REQUIRE(preflightResult.outputActions.invalid());
+
+  // Test non-existent input file
+  inputFile = "/tmp/this_does_not_exist.bin";
+  args.insertOrAssign(RawBinaryReaderFilter::k_InputFile_Key, std::make_any<FileSystemPathParameter::ValueType>(fs::path(inputFile)));
+  preflightResult = filter.preflight(ds, args);
+  REQUIRE(preflightResult.outputActions.invalid());
+
+  // Update InputFile to the correct path and write the file so it exists
+  inputFile = (fs::temp_directory_path() /= "raw_binary_reader_data.bin").string();
+  args.insertOrAssign(RawBinaryReaderFilter::k_InputFile_Key, std::make_any<FileSystemPathParameter::ValueType>(fs::path(inputFile)));
+
+  std::array<char, 10> buffer = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
+  std::fstream myFile(inputFile, std::ios::out | std::ios::binary);
+  myFile.write(buffer.data(), 10);
+
+  // Test empty input file
+  preflightResult = filter.preflight(ds, args);
+  REQUIRE(preflightResult.outputActions.invalid());
+  myFile.close();
+
+  // Test zero components
+  args.insertOrAssign(RawBinaryReaderFilter::k_NumberOfComponents_Key, std::make_any<int32>(0));
+  preflightResult = filter.preflight(ds, args);
+  REQUIRE(preflightResult.outputActions.invalid());
+  args.insertOrAssign(RawBinaryReaderFilter::k_NumberOfComponents_Key, std::make_any<int32>(1));
+
+  // Test incorrect scalar type
+  args.insertOrAssign(RawBinaryReaderFilter::k_ScalarType_Key, std::make_any<NumericType>(NumericType::uint32));
+  preflightResult = filter.preflight(ds, args);
+  REQUIRE(preflightResult.outputActions.invalid());
+  args.insertOrAssign(RawBinaryReaderFilter::k_ScalarType_Key, std::make_any<NumericType>(NumericType::uint8));
+
+  // Test incorrect components
+  args.insertOrAssign(RawBinaryReaderFilter::k_NumberOfComponents_Key, std::make_any<int32>(3));
+  preflightResult = filter.preflight(ds, args);
+  REQUIRE(preflightResult.outputActions.invalid());
+  args.insertOrAssign(RawBinaryReaderFilter::k_NumberOfComponents_Key, std::make_any<int32>(1));
+
```

Comment:
Was this all that was tested from SIMPL? I thought we tested all the primitive types using a template method?